### ISSUE TITLE
Automate Upgrade Scenarios

### ISSUE THREAD

**Issue by valaparthvi** [rancher/hosted-providers-e2e#49]
Automate tests related to rancher upgrade.
These are the tests steps that I am thinking of:
1. create and import a downstream cluster to rancher
2. upgrade the upstream cluster on which rancher is installed
3. wait until it comes up
4. ensure the downstream clusters are working as expected
5. make an update and check it worked

Blocked on: https://github.com/rancher/shepherd/pull/28
Research into how this can be done until the blocker is removed.

This automation will be helpful with k8s version support check issues, for e.g. https://github.com/rancher/gke-operator/issues/285.

Ref:
https://github.com/rancher/rancher/blob/ddbca62569b9b9c0e5ab61dca3b879e13ac981ad/tests/v2/validation/upgrade/README.md?plain=1#L10

**Comment by valaparthvi** [comment]
cc: @cpinjani 
Please feel free to add your inputs to the issue description. This is an initial draft.

**Comment by cpinjani** [comment]
Yes, I had some thoughts on its implementation and will work on this issue, if you don't mind.

**Comment by valaparthvi** [comment]
Closing the issue since it has been added. Will create a new issue for GH action.
